src/main/webapp/script.js

```diff
@@ -27,28 +39,29 @@ function loadSortingDirection() {
 
   sortingSelector.change(function() {
     localStorage.setItem('sortDir', $(this).val());
+    $('#entertainmentItemsContainer').empty();
+
     getDashboardItems();
   });
 }
 
 /**
  * Fetches entertainment items from DashboardServlet
  * to populate the Dashboard.
+ *
+ * @param { string } cursor - the cursor pointing to the page location
+ * to get the items from
  */
-function getDashboardItems() {
+function getDashboardItems(cursor = '') {
   fetch(
-      '/dashboard?cursor=' + getUrlParam('cursor') +
+      '/dashboard?cursor=' + cursor +
       '&searchValue=' + $('#searchValue').val() +
       '&sortingDirection=' + $('#sortingDirection').val())
       .then((response) => response.json())
       .then((entertainmentItemList) => {
-        const entertainmentItemsContainer = $('#entertainmentItemsContainer');
-        entertainmentItemsContainer.empty();
```

Comment:
Follow up on the discussion about the flickering issue when you moved the empty() outside of this function. You could change it so that this function takes in an additional boolean parameter `shouldRefresh`. And only call empty() here if shouldRefresh is true.

So `loadSearchValue()` and `loadSortingDirection()` call `getDashboardItems()` with `shouldRefresh = true` but `updatePagination()` doesn't 

This might fix the flickering issue

Reply:
Yeah thanks for the suggestion it worked!